civiform/civiform
# North Star question previews

### Description

When North Star is enabled, use the new North Star UI when rendering question previews.

High level overview:

- Use htmx to swap out an HTML fragment with a Thymeleaf fragment.
- Thymeleaf content is available after htmx-settle. Update text after that event.

## Release notes

n/a

### Checklist

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)


### Instructions for manual testing

1. Log in as admin
2. Click "Questions" tab
3. Create a new question or edit an existing question
4. Observe the preview pane contains North Star UI
5. Play around with the UI. Add/remove text, add/remove options, etc.
6. Expect the preview pane updates in real time

### Issue(s) this completes

Fixes #7490
